services/user/TokenSys/ui/src/App.tsx

```diff
@@ -232,6 +258,10 @@ function App() {
                     )}
                 </div>
             </form>
+            <SharedBalances
+                tokens={tokens}
+                currentUser={userName}
+            />
```

Comment:
```js
            <SharedBalances
                tokens={tokens}
                currentUser={userName}
                queryResult={sharedBalancesResult}
                refetch={async () => {
                    await wait(1000);
                    invalidateTransferHistoryQuery();
                    invalidateSharedBalancesQuery();
                }}
            />
```

(Although it would be even better if you declare that async function separately and include it there like `refetch={onTokenAction}`. Then you could even call that in the `onSubmit()` above instead of calling the two invalidate functions separately.

Also, I did find that waiting 500-1000 ms makes this more reliable, since the blockchain can take a moment to update its state.